pkg/cloudevents/transport/httpb/options.go

```diff
@@ -0,0 +1,281 @@
+package httpb
+
+import (
+	"fmt"
+	"net"
+	nethttp "net/http"
+	"net/url"
+	"strings"
+	"time"
+)
+
+// Option is the function signature required to be considered an http.Option.
+type Option func(*Transport) error
+
+// WithTarget sets the outbound recipient of cloudevents when using an HTTP
+// request.
+func WithTarget(targetUrl string) Option {
+	return func(t *Transport) error {
+		if t == nil {
+			return fmt.Errorf("http target option can not set nil transport")
+		}
+		targetUrl = strings.TrimSpace(targetUrl)
+		if targetUrl != "" {
+			var err error
+			var target *url.URL
+			target, err = url.Parse(targetUrl)
+			if err != nil {
+				return fmt.Errorf("http target option failed to parse target url: %s", err.Error())
+			}
+
+			t.Target = target
+			// TODO: it might make sense to go back to using an http request that is pre-built to allow customization.
+			//
+			//if t.Req == nil {
+			//	t.Req = &nethttp.Request{
+			//		Method: nethttp.MethodPost,
+			//	}
+			//}
+			//t.Req.URL = target
+
+			return nil
+		}
+		return fmt.Errorf("http target option was empty string")
+	}
+}
+
+//
+//// WithMethod sets the HTTP verb (GET, POST, PUT, etc.) to use
+//// when using an HTTP request.
+//func WithMethod(method string) Option {
```

Comment:
I think that `WithMethod` and `WithHeader` should be supported directly by `http.Sender`, so here you should just configure the `http.Sender`